src/nvim/lua/executor.c

```diff
@@ -132,6 +167,16 @@ static int nlua_state_init(lua_State *const lstate) FUNC_ATTR_NONNULL_ALL
   lua_setfield(lstate, -2, "stricmp");
 
   lua_setglobal(lstate, "vim");
+
+  // expose main_loop's UV loop for luv
+  lua_pushstring(lstate, "uv_loop");
+  lua_pushlightuserdata(lstate, &main_loop.uv);
```

Comment:
I think we want to avoid lightuserdata, it seems to cause trouble on some lua versions. I suppose a full userdata could be used instead?

Reply:
You would push a pointer-sized userdata, and then store the pointer at this memory. Half a year ago we removed all userdata for portability reasons #7879 #7880 , I don't think those concerns have changed.

Reply:
Understood, will change. Thanks for providing context and examples of transition from light userdata to userdata.

Reply:
https://github.com/luvit/luv/issues/252 as you linked seems relevant, luv should support pointer to existing loop.